fix: show search in description

datsfilipe opened the pull request:
Solves #37 
<!-- Adicione aqui a linking word e o número da issue, se aplicável. -->
## O que este PR faz?
Ao procurar por um post, a descrição não estava mudando. A descrição estática estava sendo usada e as vezes não incluía o termo buscado que estava presente na propriedade `full`.
## Sumário das alterações
<!-- O que mudou no projeto? Exemplo: Modifiquei a cor do texto no botão para melhor contraste e acessibilidade -->
Agora na rota `posts` a descrição é atualizada de acordo com a pesquisa, fazendo uso da propriedade `full` do post e formatando de acordo.
## Mídia
<!-- Insira aqui screenshots de antes e depois, ou somente das alterações feitas -->
Antes das alterações:
![image](https://github.com/Nick-Gabe/central-nickgabe/assets/76636791/b6ce820c-3064-4017-817d-0a4713f5fe83)
Depois das alterações:
![image](https://github.com/Nick-Gabe/central-nickgabe/assets/76636791/4d296ef6-cd63-4b5a-aa67-c80f979f7dac)

datsfilipe committed:
fix: show search in description

When searching for a post, the description was not changing.
The static description was being used and it sometimes doesn't had the
searched term that is in the `full` property. So, now in the search
route, the description is set to include the searched term, and the ellipsis.

datsfilipe committed:
refactor: do not include title in words list

datsfilipe committed:
refactor: validate if search term in present outside of title

Nick-Gabe commented on src/app/api/posts/route.ts:
```diff
@@ -7,16 +7,47 @@ export async function GET(req: Request) {
-        })
-      )
-    );
+    const filteredPosts = posts
+      .filter((post) => {
+        return (
+          regex.test(post.full) ||
+          post.full.toLowerCase().includes(search.toLowerCase())
+        );
+      })
+      // update description
+      .map((post) => {
+        if (!search) return post;
+        const fullTextWithoutTitle = post.full.replace(post.title, '');
+        if (!fullTextWithoutTitle.toLowerCase().includes(search.toLowerCase()))
+          return post;
```
Na linha 21 ele retornaria o post normal caso não incluisse a pesquisa literalmente, o lance é que a pesquisa tbm aceita regex e nesse caso ia falhar já que tecnicamente ela não "inclui" tipo um [a] literal.

![image](https://github.com/Nick-Gabe/central-nickgabe/assets/42651514/b5ae5982-31f7-4277-baa7-6f522a9b3478)

Acho que uma solução mais bonitinha é criar uma função reutilizável ali em cima que cheque ambos os métodos.

```suggestion
    const includesSearch = (text: string): boolean => {
      return (
        regex.test(text) ||
        text.toLowerCase().includes(search.toLowerCase())
      );
    }
    
    const filteredPosts = posts
      .filter((post) => includesSearch(post.full))
      // update description
      .map((post) => {
        if (!search) return post;
        const fullTextWithoutTitle = post.full.replace(post.title, '');
        if (!includesSearch(fullTextWithoutTitle)) return post
```

Nick-Gabe reviewed (changes requested):
Bixo, que task hein aushuasuha, essa era complicada de fazer, muito obrigado pela contribuição ^^.
Comentei só sobre um edge case que podia rolar ali, mas todo o resto tá muito bom. :)

datsfilipe committed:
fix: add includeSearch function to check for the search term

Accepting reviewer suggestion.

Co-authored-by: Nícolas Gabriel <nickygamer023@gmail.com>

datsfilipe committed:
fix: prettier problems

datsfilipe committed:
fix: use includeSearch to find search term index

datsfilipe committed:
refactor: more words in beginning if text is too short

Also avoid a negative start.

datsfilipe committed:
fix: ellipsis in the beginning not working as expected

Added a verification for the first word, so if it's an emoji, a new line char or a whitespace, it will not add the ellipsis. Also if index is 0, it will not add the ellipsis.

Nick-Gabe reviewed (approved):
LGTM! :D
